lib/switchtec.c

```diff
@@ -1333,36 +1340,58 @@ static int log_c_to_file(struct switchtec_dev *dev, int sub_cmd_id, int fd)
 	return 0;
 }
 
+static int log_ram_flash_to_file(struct switchtec_dev *dev,
+				 int gen5_cmd, int gen4_cmd, int gen4_cmd_lgcy,
+				 int fd, FILE *log_def_file,
+				 struct switchtec_log_file_info *info)
+{
+	int ret;
+
+	if (switchtec_is_gen5(dev)) {
+		return log_a_to_file(dev, gen5_cmd, fd, log_def_file,
+				     info);
+	} else {
+		ret = log_a_to_file(dev, gen4_cmd, fd, log_def_file,
+				    info);
+		if (!ret)
```

Comment:
I still think we should only try the legacy command if this failure indicates a bad MRPC command.

Reply:
Somehow hardware gives random error number if this new subcommand is not supported. So checking for all errors is the safe way. 

I've added comments to reflect this limitation.

Reply:
What? That seems like hardware is really broken if it returns a random error number! That needs to be fixed ASAP.

But this is not safe. If it fails for some other legitimate reason then it falls back on a command that may or may not exist, then the reason for the failure is masked.... That is not a good idea.

Reply:
Sorry about the confusion, by 'random' I mean the error code returned is not the expected 'Invalid subcommand' error code, not that it returns a different number on each run. 

Yes there could be other issues masked if we simply check for non-zero. 

I have updated the code to check for the exact error code returned, and updated the comments.

```
		/* somehow hardware returns ERR_LOGC_PORT_ARDY_BIND
		 * instead of ERR_SUBCMD_INVALID if this subcommand
		 * is not supported, so we fall back to legacy
		 * subcommand on ERR_LOGC_PORT_ARDY_BIND error */
```